File: admin/src/main/java/com/weicoder/admin/po/Menu.scala
package com.weicoder.admin.po

import javax.persistence.Entity
import org.hibernate.annotations.DynamicInsert
import org.hibernate.annotations.DynamicUpdate
import org.springframework.context.annotation.Scope
import org.springframework.stereotype.Component
import org.springframework.beans.factory.config.ConfigurableBeanFactory
import com.weicoder.base.annotation.Cache
import com.weicoder.site.entity.base.BaseEntityId
import scala.beans.BeanProperty

/**
 * 菜单实体
 * @author WD
 * @since JDK7
 * @version 1._ 2__9-11-23
 */
@Component
@Scope(ConfigurableBeanFactory.SCOPE_PROTOTYPE)
@Entity
@DynamicInsert
@DynamicUpdate
@Cache
class Menu extends BaseEntityId {
  // 上级菜单ID
  @BeanProperty
  var menuId: Integer = _
  // 链接
  @BeanProperty
  var url: String = _
  // 名称
  @BeanProperty
  var name: String = _
}

File: admin/src/main/java/com/weicoder/admin/po/Menu.java
package com.weicoder.admin.po;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;

import lombok.Data;

/**
 * 菜单实体
 * 
 * @author WD
 */
@Entity
@Data
public class Menu {
	// ID
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private int id;
	// 上级菜单ID
	private Integer menuId;
	// 链接
	private String url;
	// 名称
	private String name;
}
